src/main/java/studentlist/StudentList.java

```diff
@@ -53,47 +114,160 @@ public int getNumStudents() {
         return students.size();
     }
 
-    public Student makeStudent(String input) throws Exception {
-        // Split based on id/, gpa/, and p/ without spaces
-        String[] parts = input.split("id/|gpa/|p/");
+    /**
+     * Handles the creation of a Student object by validating the input for Student ID, GPA, and preferences.
+     *
+     * @param input The input string containing student information.
+     * @return The created Student object if all validations pass.
+     * @throws SEPException If any validation errors occur, a SEPException containing all error messages is thrown.
+     */
+    public Student makeStudent(String input) throws SEPException {
+        Set<String> errorMessages = new HashSet<>();
```

Comment:
Interesting use of `HashSet` as a `Set` to hold the exception messages

Reply:
Thanks, I used a hashset to prevent duplicate error messages when validating the format of preference rankings